docs/groovy/conceptual/formula-validation-configuration.md

```diff
@@ -3,93 +3,265 @@ title: Formula validation configuration
 
 ```java
-public static class AnnotatedMethods {
-    @UserInvocationPermitted(value = "example_method_annotation1")
-    public String im() {
-        return "im";
+public class MixedUtilities {
+    @UserInvocationPermitted(value = "safe_math")
+    public static double safeDivide(double a, double b) {
+        return b != 0 ? a / b : 0;  // Available in formulas
+    }
+
+    @UserInvocationPermitted(value = "text_processing")
+    public String cleanText(String input) {
+        return input.trim().toLowerCase();  // Available in formulas
     }
 
-    @UserInvocationPermitted(value = "example_method_annotation2")
-    public static String stm() {
-        return "sm";
+    public void dangerousOperation() {
+        // No annotation = NOT available in formulas
     }
 }
 ```
 
-The best practice is to group related methods into a single named set, to avoid excess configuration. In addition to annotating the classes, you must set a property of the form `ColumnExpressionValidator.annotationSets.<name>`. The following property permits the methods annotated in the examples above:
+### Best practices for annotations
+
+When designing your annotations, follow these recommendations:
+
+- **Use descriptive annotation values** that clearly indicate the method category (e.g., `math_operations`, `string_utilities`)
+- **Group logically related methods** together rather than creating separate annotations for each method
+- **Avoid creating too many small annotation categories**, as this increases configuration complexity
+- **Consider the security implications** - only annotate methods that are safe for user formulas
+- **Group related methods under the same annotation value** to reduce configuration
+
+### Configuration
+
+After adding annotations to your code, you must tell Deephaven to recognize them by setting configuration properties.
+
+**Default configuration:**
+
+Deephaven comes with pre-configured annotations for built-in functionality:
 
 ```properties
-ColumnExpressionValidator.annotationSets.example=example_instance_methods,example_static_methods,example_method_annotation1,example_method_annotation2
+ColumnExpressionValidator.annotationSets.default=base,vector,function_library
 ```
 
-The default configuration for Deephaven libraries is:
+**Adding your own annotations:**
+
+- The property name format is: `ColumnExpressionValidator.annotationSets.<your_name>`.
+- List all your annotation values separated by commas.
 
 ```properties
-ColumnExpressionValidator.annotationSets.default=base,vector,function_library
+# Enable the annotation categories you defined
+ColumnExpressionValidator.annotationSets.myapp=static_math_methods,string_helpers,safe_math,text_processing
 ```
 
-## Pointcut expressions
+## Method 2: Pointcut expressions (for external libraries)
+
+Use this approach when you want to allow methods from libraries you don't control (like Java's built-in classes, third-party libraries, etc.).
```

Comment:
You need to restore the link to openrewrite which has the comprehensive documentation on their format.  Telling people what the standard format is and the actual documentation is important for them to configure things correctly.